### ISSUE TITLE
Cannot import dotnet 8 assembly in javascript 

### ISSUE THREAD

**Issue by nojaf** [microsoft/node-api-dotnet#190]
Hello,

I'm playing around with this project and I am trying to follow the steps in https://github.com/microsoft/node-api-dotnet/blob/main/Docs/node-module.md.

```xml
<Project Sdk="Microsoft.NET.Sdk">

    <PropertyGroup>
        <OutputType>Library</OutputType>
        <TargetFramework>net8.0</TargetFramework>
        <RollForward>Major</RollForward>
    </PropertyGroup>

    <ItemGroup>
        <Compile Include="Foo.fs" />
    </ItemGroup>
    
    <ItemGroup>
        <PackageReference Include="Microsoft.JavaScript.NodeApi" />
        <PackageReference Include="Microsoft.JavaScript.NodeApi.Generator" />
    </ItemGroup>
</Project>
```

At first I was unable to build this project:

```
App: /home/nojaf/.nuget/packages/microsoft.javascript.nodeapi.generator/0.4.27/analyzers/dotnet/cs/Microsoft.JavaScript.NodeApi.Generator.dll
  Architecture: x64
  Framework: 'Microsoft.NETCore.App', version '6.0.0' (x64)
  .NET location: /home/nojaf/.dotnet/
  
  The following frameworks were found:
    7.0.5 at [/home/nojaf/.dotnet/shared/Microsoft.NETCore.App]
    7.0.8 at [/home/nojaf/.dotnet/shared/Microsoft.NETCore.App]
    7.0.10 at [/home/nojaf/.dotnet/shared/Microsoft.NETCore.App]
    8.0.0-preview.7.23375.6 at [/home/nojaf/.dotnet/shared/Microsoft.NETCore.App]
    8.0.0-rc.1.23419.4 at [/home/nojaf/.dotnet/shared/Microsoft.NETCore.App]
    8.0.0-rc.2.23479.6 at [/home/nojaf/.dotnet/shared/Microsoft.NETCore.App]
    8.0.0 at [/home/nojaf/.dotnet/shared/Microsoft.NETCore.App]
  
  Learn more:
  https://aka.ms/dotnet/app-launch-failed
  
  To install missing framework, download:
  https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=6.0.0&arch=x64&rid=linux-x64&os=ubuntu.22.04
```

After installing the runtime, that got resolved but I was hoping I could roll forward the usage of `Microsoft.NETCore.App` to 8.0.0. Is it possible to do this?

Next, I'm unable to load my assembly in Node:

```js
import dotnet from 'node-api-dotnet';
const dll = "/MyAssembly.dll"
dotnet.load(dll);
```

```
node:internal/process/esm_loader:97
    internalBinding('errors').triggerUncaughtException(
                              ^

Error: ReflectionTypeLoad_LoadFailed
Could not load file or assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. An operation is not legal in the current state. (0x80131509)
Could not load file or assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. An operation is not legal in the current state. (0x80131509)
Could not load file or assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. An operation is not legal in the current state. (0x80131509)
Could not load file or assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. An operation is not legal in the current state. (0x80131509)
Could not load file or assembly 'System.Runtime, Version=8.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'. An operation is not legal in the current state. (0x80131509)
    at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
    at System.Reflection.Assembly.GetTypes()
    at Microsoft.JavaScript.NodeApi.DotNetHost.ManagedHost.LoadAssemblyTypes(Assembly assembly)
    at Microsoft.JavaScript.NodeApi.DotNetHost.ManagedHost.LoadAssembly(String assemblyNameOrFilePath, Boolean allowNativeLibrary)
    at Microsoft.JavaScript.NodeApi.DotNetHost.ManagedHost.LoadAssembly(JSCallbackArgs args)
    at Microsoft.JavaScript.NodeApi.JSNativeApi.InvokeCallback[TDescriptor](napi_env env, napi_callback_info callbackInfo, JSValueScopeType scopeType, Func`2 getCallbackDescriptor)
    at file:///home/nojaf/projects/vite-plugin-fable/kh.js:3:8
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Node.js v18.16.1
```

This again seems like something I would like some roll-forward mechanism to solve.

And my last question would be whether C# is a hard requirement for this to work.
I'm trying my luck with F# right now and I'm bumping into

```
node:internal/process/esm_loader:97
    internalBinding('errors').triggerUncaughtException(
                              ^

Error: Failed to load module. A module initialize method was not found.
    at Microsoft.JavaScript.NodeApi.DotNetHost.ManagedHost.LoadModule(JSCallbackArgs args)
    at Microsoft.JavaScript.NodeApi.JSNativeApi.InvokeCallback[TDescriptor](napi_env env, napi_callback_info callbackInfo, JSValueScopeType scopeType, Func`2 getCallbackDescriptor)
```

Would I be able to do all this in F#?

**Comment by nojaf** [comment]
Alright, I think I'm able to answer my question when it comes to F# being supported.
The answer would be no, as I see that the generated `Microsoft.JavaScript.NodeApi.Generated` is not present in my F# assembly.

I see this in my C# attempt:

![image](https://github.com/microsoft/node-api-dotnet/assets/2621499/410e06c5-1912-49dc-9667-910f05488632)

And nothing similar is present for F#.

**Comment by jasongin** [comment]
The `Microsoft.JavaScript.NodeApi.Generator` package is a [C# source-generator](https://learn.microsoft.com/en-us/dotnet/csharp/roslyn-sdk/source-generators-overview) so it will only work with a C# project. That generator looks for `[JSExport]` and related attributes and generates code to marshal calls to/from JS. It also generates the module initialize method, which is what was missing in your last error.


The APIs in the `Microsoft.JavaScript.NodeApi` package are not C#-specific, so you should be able to use them from F#. However it won't be as easy without the generator, as some initialization and marshalling details are tricky.

We have talked about adding an example project that doesn't use the generator, but we don't have anything yet. I guess you could see what kind of code gets generated in C# and try to translate to F#.

**Comment by jasongin** [comment]
> Could not load file or assembly 'System.Runtime, Version=8.0.0.0

The `import dotnet from 'node-api-dotnet'` statement currently loads .NET 6 by default. You're supposed to be able to specify the .NET version via `import dotnet from 'node-api-dotnet/net8.0'`, but it looks like that isn't working right now. I'll look into it.

**Comment by jasongin** [comment]
> You're supposed to be able to specify the .NET version via import dotnet from 'node-api-dotnet/net8.0', but it looks like that isn't working right now.

Fixed by #193
